src/main/java/seedu/address/model/task/Title.java

```diff
@@ -0,0 +1,68 @@
+package seedu.address.model.task;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.Objects;
+
+/**
+ * Represents a Task's title in the task list.
+ * Guarantees: immutable; is valid as declared in {@link #isValidTitle(String)}
+ */
+public class Title {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Title should only contain alphanumeric characters and spaces, and it should not be blank";
+
+    /*
+     * The first character of the task title must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[\\p{Alnum}][\\p{Alnum} ]*";
+
+    public final String taskTitle;
```

Comment:
Similar naming concern here. Or perhaps you can think of a more descriptive name that doesn't point towards the higher class.

```suggestion
    public final String value;
```

Reply:
Resolved in https://github.com/AY2324S1-CS2103T-T10-2/tp/pull/81/commits/e92f537770862854c44c9f42c63beebf730d0639